install/0000_80_machine-config-operator_09_etcddefragmentation_cronjob.yaml

```diff
@@ -0,0 +1,55 @@
+---
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: etcd-defrag
+  namespace: openshift-etcd
+spec:
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          hostNetwork: true
+          containers:
+            - command: ["/usr/bin/sh"]
+              terminationMessagePolicy: FallbackToLogsOnError
+              args:
+                - "-c"
+                - |
+                  set -euo pipefail
+
+                  export ETCDCTL_API=3
+                  export ETCDCTL_CACERT=/etc/ssl/etcd/ca.crt ETCDCTL_CERT=$(find /etc/ssl/ -name *peer*crt) ETCDCTL_KEY=$(find /etc/ssl/ -name *peer*key)
+                  export ETCDCTL_ENDPOINTS=https://127.0.0.1:2379;
+                  for m in $(etcdctl member list); do
+                    [[ $m =~ "2379" ]] && echo "Defragmenting endpoint $m" && etcdctl --endpoints=$m defrag
```

Comment:
To simplify this why not create the array with only valid client records? `etcdctl member list | awk '{ print $5}' | uniq`

Reply:
In a larger db this will timeout how are you going to handle that to cover all circumstances?

Reply:
How will we handle retry if we can not reach a member in the list? If the first defrag fails we exit now right?

Reply:
IS there a way we can alert admin if cronjob fails I know with traditional cron we could do something like.

```
MAILTO=admin@cluster.com
```

If this fails we should alert somehow. Not email:)

Reply:
reminder `restartPolicy: Never`

Reply:
1. I don't believe we get to see the alerts, good feature to suggest upstream. 
2. I am not sure how to handle timeouts on large db, maybe set the client timeout to 10 minutes?
3. IMO we shouldn't handle retry if one fails for some reason. We skip the defrag of that member and try again next week
4. IMO we should run the defrag command on each member independent of one another and space it out by 30 seconds as discussed on slack.